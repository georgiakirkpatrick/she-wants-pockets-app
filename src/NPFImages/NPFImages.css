.NPFImages__fieldset {
    position: relative;
    width: 100%;
    padding: 10px;
    margin: 5px 0;
    background-color: rgb(238, 238, 238);
    border: none;
}