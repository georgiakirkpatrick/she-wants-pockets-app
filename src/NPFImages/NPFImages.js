import React from 'react'
import '@fortawesome/react-fontawesome'
import FormPage from '../FormPage/FormPage'
import FormButton from '../FormButton/FormButton'
import FormFieldset from '../FormFieldset/FormFieldset'
import FormPromptWithSub from '../FormPromptWithSub/FormPromptWithSub'
import FormUrlInput from '../FormUrlInput/FormUrlInput'
import NPFFooter from '../NPFFooter/NPFFooter';
import formData from '../FORM_DATA'

const NPFImages = props => {
    const addAnImage = (index) => {
        const fieldsetToUpdate = props.colorFieldsets[index]
        fieldsetToUpdate.colorImageUrls.push('')
        props.setColorFieldsets([ ...props.colorFieldsets ])    
    }

    const changeSwatchInput = (index, event) => {
        const values = [...props.colorFieldsets]
        values[index][event.target.name] = event.target.value
        props.setColorFieldsets(values)
    }

    const changeImageInput = (index, i, e) => {
        const values = [...props.colorFieldsets]
        values[index][e.target.name][i] = e.target.value
    }

    return (
        <div>
            <FormPage title={formData.images.pageTitle}>
                <FormPromptWithSub prompt={formData.images.pagePrompts.prompt} promptSubtitle={formData.images.pagePrompts.promptSubtitle} />
                {props.colorFieldsets.map((colorFieldset, index) => (
                    <FormFieldset key={index} className='NPFImages__fieldset'>
                        <FormPromptWithSub prompt={colorFieldset.colorName} promptSubtitle='' />
                        <FormUrlInput 
                            id={'color-swatch-url-fieldset' + index}
                            name='colorSwatchUrlFieldset'
                            prompt={formData.images.colorSwatchUrlFieldset.prompt}
                            name={formData.images.colorSwatchUrlFieldset.name}
                            currentValue={colorFieldset.colorSwatchUrl}
                            handleChange={event => {
                                changeSwatchInput(index, event)
                            }}
                        />
                        {colorFieldset.colorImageUrls.map((colorImageUrl, i) => (
                            <FormUrlInput
                                key={i}
                                id={'color-image-urls-' + i}
                                name='colorImageUrls'
                                prompt={formData.images.imageUrl.prompt}
                                currentValue={colorImageUrl}
                                handleChange={e => {
                                    changeImageInput(index, i, e)
                                }}
                            />
                        ))}
                        
                        <FormButton
                            buttonText='ADD AN IMAGE FOR THIS COLOR'
                            handleClick={() => addAnImage(index)}
                        />
                   </ FormFieldset>
                ))}
            </FormPage>

            <NPFFooter 
                buttons='prevNext' 
                previousButton={() => props.setPage(props.currentPage - 1)} 
                nextButton={() => {props.setPage(props.currentPage + 1)}}
            />
        </div>
    )
}

NPFImages.defaultProps = {
    colorFieldsets: [
        { 
            colorName: '', 
            colorDescription: '1' ,
            colorSwatchUrl: '',
            colorImageUrls: ['']
        }
    ],
    currentPage: 0,
    setPage: () => {},
    setColorFieldsets: () => {}
}

export default NPFImages