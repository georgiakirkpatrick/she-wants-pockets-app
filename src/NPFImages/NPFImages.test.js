import React from 'react'
import ReactDOM from 'react-dom'
import renderer from 'react-test-renderer'
import NPFImages from './NPFImages'

describe('NPFImages', () => {
    it ('renders without crashing', () => {
        const div = document.createElement('div')
        ReactDOM.render(<NPFImages />, div)
        ReactDOM.unmountComponentAtNode(div)
    })

    it ('render the UI as expected', () => {
        const tree = renderer
            .create(<NPFImages />)
            .toJSON()
        expect(tree).toMatchSnapshot()
    })
})