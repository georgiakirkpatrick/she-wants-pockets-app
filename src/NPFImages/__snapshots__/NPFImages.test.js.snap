// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NPFImages render the UI as expected 1`] = `
<div>
  <section
    className="FormPage"
  >
    <h1>
      Images
    </h1>
    <form>
      <div
        className="FormPromptWithSub"
      >
        <p
          className="FormPromptWithSub__prompt"
        >
          Enter the color swatch URL and image URL(s) for each color
        </p>
        <p
          className="FormPromptWithSub__subtitle"
        >
          
        </p>
      </div>
      <fieldset
        className="FormFieldset"
      >
        <div />
        <div
          className="FormPromptWithSub"
        >
          <p
            className="FormPromptWithSub__prompt"
          >
            
          </p>
          <p
            className="FormPromptWithSub__subtitle"
          >
            
          </p>
        </div>
        <div
          className="FormUrlInput"
        >
          <label
            htmlFor="color-swatch-url-fieldset0"
          >
            Color swatch URL
          </label>
          <input
            defaultValue=""
            id="color-swatch-url-fieldset0"
            name="colorSwatchUrl"
            onChange={[Function]}
            type="url"
          />
        </div>
        <div
          className="FormUrlInput"
        >
          <label
            htmlFor="color-image-urls-0"
          >
            Image URL for this color
          </label>
          <input
            defaultValue=""
            id="color-image-urls-0"
            name="colorImageUrls"
            onChange={[Function]}
            type="url"
          />
        </div>
        <button
          className="FormButton"
          onClick={[Function]}
          type="button"
        >
          ADD AN IMAGE FOR THIS COLOR
        </button>
      </fieldset>
    </form>
  </section>
  <footer
    className="NPFFooter"
  >
    <button
      className="NPFFooter_previous"
      onClick={[Function]}
    >
      <svg
        aria-hidden="true"
        className="svg-inline--fa fa-arrow-left fa-w-14 NPFFooter_arrow"
        data-icon="arrow-left"
        data-prefix="fas"
        focusable="false"
        role="img"
        style={Object {}}
        viewBox="0 0 448 512"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M257.5 445.1l-22.2 22.2c-9.4 9.4-24.6 9.4-33.9 0L7 273c-9.4-9.4-9.4-24.6 0-33.9L201.4 44.7c9.4-9.4 24.6-9.4 33.9 0l22.2 22.2c9.5 9.5 9.3 25-.4 34.3L136.6 216H424c13.3 0 24 10.7 24 24v32c0 13.3-10.7 24-24 24H136.6l120.5 114.8c9.8 9.3 10 24.8.4 34.3z"
          fill="currentColor"
          style={Object {}}
        />
      </svg>
      <span>
        PREVIOUS
      </span>
    </button>
    <button
      className="NPFFooter_next"
      onClick={[Function]}
    >
      <span>
        NEXT
      </span>
      <svg
        aria-hidden="true"
        className="svg-inline--fa fa-arrow-right fa-w-14 NPFFooter_arrow"
        data-icon="arrow-right"
        data-prefix="fas"
        focusable="false"
        role="img"
        style={Object {}}
        viewBox="0 0 448 512"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M190.5 66.9l22.2-22.2c9.4-9.4 24.6-9.4 33.9 0L441 239c9.4 9.4 9.4 24.6 0 33.9L246.6 467.3c-9.4 9.4-24.6 9.4-33.9 0l-22.2-22.2c-9.5-9.5-9.3-25 .4-34.3L311.4 296H24c-13.3 0-24-10.7-24-24v-32c0-13.3 10.7-24 24-24h287.4L190.9 101.2c-9.8-9.3-10-24.8-.4-34.3z"
          fill="currentColor"
          style={Object {}}
        />
      </svg>
    </button>
  </footer>
</div>
`;
